tests/conformance/conformance_test.go

```diff
@@ -3003,6 +3003,26 @@ var internalTestCases = []testCase{
 		dockerUseBuildKit: true,
 	},
 
+	{
+		name:              "heredoc-copy",
+		dockerfile:        "Dockerfile.heredoc_copy",
+		dockerUseBuildKit: true,
+		contextDir:        "heredoc",
+		fsSkip: []string{"(dir):test:mtime",
+			"(dir):test2:mtime",
+			"(dir):test:(dir):humans.txt:mtime",
+			"(dir):test:(dir):robots.txt:mtime",
+			"(dir):test:(dir):file:mtime",
+			"(dir):test:(dir):file:uid", /* buildkit's regular COPY sets UID which is inconsistent with docker and buildah*/
+			"(dir):test:(dir):file:gid", /* buildkit's regular COPY sets GID which is inconsistent with docker and buildah*/
+			"(dir):test2:(dir):humans.txt:mtime",
+			"(dir):test2:(dir):robots.txt:mtime",
+			"(dir):test2:(dir):image_file:mtime",
+			"(dir):test2:(dir):image_file:uid", /* buildkit's regular COPY sets UID which is inconsistent with docker and buildah*/
+			"(dir):test2:(dir):image_file:gid", /* buildkit's regular COPY sets GID which is inconsistent with docker and buildah*/
```

Comment:
What values does it use?  Why is it okay to ignore the differences?

Reply:
Buildkit sets `uid` and `gid` to `1000` with `COPY` instruction in conformance test and `buildah/docker` keeps it 0, i suspect either problem is with buildkit client in conformance test or could be a buildkit issue, this happens on upstream as well so I was not able to spend time on it figure out root cause for it. Since this was not related to this PR hence I did not touch it for now  ( will investigate issue in a different PR )

**Issue reproducer for upstream:**

Following problem is with all the `COPY` ( where `uid` and `gid` are not modified ) related tests included in conformance_test.go

For example a try out example is
```diff
diff --git a/tests/conformance/conformance_test.go b/tests/conformance/conformance_test.go
index 6ec8f3837..e494cb116 100644
--- a/tests/conformance/conformance_test.go
+++ b/tests/conformance/conformance_test.go
@@ -2784,6 +2784,7 @@ var internalTestCases = []testCase{
        {
                name:       "copy-for-user", // flip side of issue #2518
                contextDir: "copy",
+               dockerUseBuildKit: true,
                dockerfileContents: strings.Join([]string{
                        `FROM alpine`,
                        `USER 66:66`,
```

and output fails

```console
[fl@fedora buildah]$ sudo GOROOT=/usr/local/go go test -v -timeout=30m -run TestConformance/copy-for-user ./tests/conformance
=== RUN   TestConformance
=== RUN   TestConformance/copy-for-user
    conformance_test.go:482: 
        	Error Trace:	/home/fl/heredoc-exp/buildah/tests/conformance/conformance_test.go:482
        	            				/home/fl/heredoc-exp/buildah/tests/conformance/conformance_test.go:333
        	            				/home/fl/heredoc-exp/buildah/tests/conformance/conformance_test.go:151
        	Error:      	Filesystem contents differ
        	Test:       	TestConformance/copy-for-user
        	Messages:   	File attributes in both versions have different values:
        	            	File:attr          Docker        buildah
        	            	/script:uid        1000          0
        	            	/script:gid        1000          0
    conformance_test.go:355: Context "/home/fl/heredoc-exp/buildah/tests/conformance/testdata/copy"
    conformance_test.go:367: Dockerfile contents:
        FROM alpine
        USER 66:66
        COPY /script /script
        (no final end-of-line)
=== RUN   TestConformance/copy-for-user-with-chown
--- FAIL: TestConformance (13.23s)
    --- FAIL: TestConformance/copy-for-user (6.97s)
    --- PASS: TestConformance/copy-for-user-with-chown (6.26s)
FAIL
FAIL	github.com/containers/buildah/tests/conformance	13.244s
FAIL
```

Reply:
So BuildKit is retaining the uid/gid of the files in the context directory, which are owned by your unprivileged user, instead of setting them to 0?

Reply:
I guess it's expecting the client to hand it an archive with ownerships set to 0:0 and go-dockerclient isn't doing that.

Reply:
Opened https://github.com/containers/buildah/pull/5171 to generate the buildcontext archive we hand to dockerd with imagebuilder's helper, which sets ownerships in it to 0:0.

Reply:
Thanks I verified against #5171 and it works, I have removed `uid` and `gid` skips, will rebase after #5171